CF-115 axios default settings

gtangelo opened the pull request:
Included the use of axios default settings for the base url so we can easily change it later if needed (i.e. If deployment version uses a different basurl/port instead of localhost:8000)

gtangelo committed:
CF-115 axios default settings

lhjt commented on frontend/src/axios.js:
```diff
@@ -0,0 +1,4 @@
+import axios from 'axios';
+
+axios.defaults.baseURL = 'http://localhost:8000/';
```
It'd probably be better to have this as a configurable environment variable instead that is set during build time. This means that we can set the appropriate URL automatically into the production container build pipelines.

Snuzzn committed:
set up env variable for api base url

Snuzzn committed:
Update README.md

Snuzzn reviewed (approved):
Looks good! I updated the docker-compose file to take in a new environment variable for the base url (instructions in readme). If this works for you too, merge it in and then I'll let everyone know in general chat to create that env file.

Snuzzn committed:
added base url to CI teting

Snuzzn committed:
Merge branch 'gabriel/fix/axios-settings' of github.com:csesoc/Circles into gabriel/fix/axios-settings

lhjt commented on README.md:
```diff
@@ -21,6 +21,9 @@ In `mongodb.env`, add:
 - `MONGO_INITDB_ROOT_PASSWORD=...`
 
+In `frontend.env`, add:
+- `REACT_APP_BASE_URL=http://localhost:8000/`
```
I'm glad that you have included this as a required environment variable; however, could you add a small sentence/description as to what this should represent? Based on this, it seems to be the URL that the react app is being hosted on, rather than necessarily the API endpoint (for example if the API is running on a different domain).

lhjt replied:
So include something like 
```suggestion
- `REACT_APP_BASE_URL=http://localhost:8000/` (this should be ...)
```

gtangelo committed:
renamed frontend env variable and added fallback to base url for axios

gtangelo commented:
Renamed the `REACT_APP_BASE_URL` to `BACKEND_API_BASE_URL` to make it more clear and added a brief description as what Jared suggested. Also, in the axios configuration, added `http://localhost:8000/` as a fallback base url for the api call in case the environment variable is not specified. :)

Peedee2002 reviewed (approved):
Nice, i like this! Just make a quick ping in general to tell everyone to add that env file when you merge this in.

lhjt reviewed (approved):
Looks good 👍
